### PR TITLE
syncutil: remove AtomicBool

### PR DESCRIPTION
**Pull request by RaduBerinde**
This type now duplicates `atomic.Bool`.

Epic: none
Release note: None

### CODE REVIEW

File: pkg/kv/kvserver/store_gossip.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/redact"
+	"go.uber.org/atomic"
```
Comment (stevendanna) [inline pkg/kv/kvserver/store_gossip.go:35]: Was this intentional? Seems like we probably want `sync.Atomic`

Comment (RaduBerinde) [reply pkg/kv/kvserver/store_gossip.go]: Oh, nice catch! That was Goland. There are a bunch more, fixed all of them.

**Review by stevendanna — APPROVED** [review]

**Review by nvb — APPROVED** [review]
:lgtm:

> Reviewed 20 of 44 files at r1, 24 of 24 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/125614#-:-O-H3bpfA9Xl8_lH7Wmb:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @abarganier and @stevendanna)

___
*[`pkg/kv/kvserver/store_gossip.go` line 35 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/125614#gh-1637598042:-O-H36EBAru1ORepjx6b:b8vhowf) ([raw file](https://github.com/cockroachdb/cockroach/blob/9add91ba04729e05a28b14cd2d76a408a3001f47/pkg/kv/kvserver/store_gossip.go#L35)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

Oh, nice catch! That was Goland. There are a bunch more, fixed all of them.
</blockquote></details>

I've seen Goland do this a few times. Mind adding the "go.uber.org/atomic" to "atomic" mapping to `TestForbiddenImports`?

**Comment by RaduBerinde** [comment]
*[`pkg/kv/kvserver/store_gossip.go` line 35 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/125614#gh-1637598042:-O-H4qbP4UDrscZfhV6p:b-fznoh6) ([raw file](https://github.com/cockroachdb/cockroach/blob/9add91ba04729e05a28b14cd2d76a408a3001f47/pkg/kv/kvserver/store_gossip.go#L35)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

I've seen Goland do this a few times. Mind adding the "go.uber.org/atomic" to "atomic" mapping to `TestForbiddenImports`?
</blockquote></details>

Good idea, done!

**Review by RaduBerinde — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/125614)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @abarganier and @stevendanna)

___
*[`pkg/kv/kvserver/store_gossip.go` line 35 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/125614#gh-1637598042:-O-H55AvCIdNj_OgepEN:b-w4g36g) ([raw file](https://github.com/cockroachdb/cockroach/blob/9add91ba04729e05a28b14cd2d76a408a3001f47/pkg/kv/kvserver/store_gossip.go#L35)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

Good idea, done!
</blockquote></details>

Oh, I see now why. The existing code used `Get()/Set()` and the uber variant has those. So it was actually being pretty smart.

**Review by sean- — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/125614)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @abarganier and @stevendanna)

**Pull request merged**